packages/merge-on-green/src/merge-on-green.ts

```diff
@@ -246,11 +309,58 @@ function handler(app: Application) {
           state: 'continue',
           url: context.payload.pull_request.html_url,
           branchProtection: branchProtection,
+          label,
+          author,
+          reactionId,
+          installationId,
         },
         context.payload.pull_request.html_url
       );
     }
   });
+
+  app.on(
+    ['pull_request.unlabeled', 'pull_request.closed', 'pull_request.merged'],
+    async context => {
+      const prNumber = context.payload.pull_request.number;
+      const owner = context.payload.repository.owner.login;
+      const repo = context.payload.repository.name;
+
+      // check to see if the label is on the PR
+      const label = context.payload.pull_request.labels.find(
+        (label: Label) =>
+          label.name === MERGE_ON_GREEN_LABEL ||
+          label.name === MERGE_ON_GREEN_LABEL_SECURE
+      )?.name;
+
+      // If the label is on the PR but the action was unlabeled, it means the PR had some other
+      // label removed. No action needs to be taken.
+      if (label && context.payload.action === 'unlabeled') {
+        logger.info(
+          `correct label ${label} is still on ${repo}/${prNumber}, will continue watching`
+        );
+        return;
+      }
+
+      // Check to see if the PR exists in the table before trying to delete. We also
+      // need to do this to get the reaction id to remove the reaction when MOG is finished.
+      const PR: WatchPR = await handler.getPR(
```

Comment:
looks like we're using lower camel-case for variables. we should also try to avoid abbreviations if possible.
```suggestion
      const watchedPullRequest: WatchPR = await handler.getPR(
```